api/caasunitprovisioner/client.go

```diff
@@ -130,6 +130,7 @@ type ProvisioningInfo struct {
 	PodSpec     string
 	Constraints constraints.Value
 	Filesystems []storage.KubernetesFilesystemParams
+	Devices     []params.KubernetesDeviceParams // ??? combine with .Constrains ???
```

Comment:
Device constraints are separate to other constraints.
As with storage, we pass across params to specify what devices are needed.